src/vs/workbench/contrib/terminal/browser/terminalLinkHandler.ts

```diff
@@ -66,14 +68,32 @@ interface IPath {
 	normalize(path: string): string;
 }
 
+class TerminalLinkAdapter implements ILinkComputerTarget {
+	constructor(
+		private _xterm: Terminal
+	) { }
+
+	getLineCount(): number {
+		return this._xterm.buffer.length;
+	}
+
+	getLineContent(lineNumber: number): string {
+		return this._xterm.buffer.getLine(lineNumber)?.translateToString() || '';
+	}
+}
```

Comment:
It would be preferable to not modify `LinkComputer` at all, could this adapter instead pull the wrapped lines and essentially pretend to be the terminal's viewport?

```ts
class TerminalLinkAdapter implements ILinkComputerTarget {
    constructor(
        private _xterm: Terminal,
        private lineStart: number,
        private lineEnd
        private bufferLines
    ) {
        // Compute wrapped lines from lineStart to lineEnd in the buffer
    }

    getLineCount(): number {
        // Return the number of _wrapped_ lines between lineStart and lineEnd
    }

    getLineContent(lineNumber: number): string {
        // Return the _wrapped_ line at index lineNumber
    }

    getWrappedLineCount(): number {
        // Return the wrapped line count, to be used when calling into LinkComputer
    }
}
```

Reply:
I thought about this, but then we have to convert the range of the provided link back to the buffer coords since the `LinkComputer` would be using our pretend viewport to calculate the range, correct?